### PR TITLE
Label Modal doesn't automatically close once the label is created

### PR DESCRIPTION
**Pull request by emettely**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

Bug fix - label creation should have closed the modal automatically, but because `trackEvent` was not defined (not passed from parent component that had the analytics context), when it was called after label creation, it threw an exception.

**Describe what the PR does**    
- [x] update components to have access to trackEvent where they did not have access to analytics context.
- [x] update props 

**State whether the PR is ready for review or whether it needs extra work**    
Ready for review

**Additional context**

### CODE REVIEW

File: src/Components/PaperEditor/TranscriptsContainer/index.js
Comment (emettely) [inline src/Components/PaperEditor/TranscriptsContainer/index.js:34]: Must pass trackEvent to child components, that don't have direct access to analytics context.
The parent components access the context via:
`export default withAuthorization(condition)(ComponentName);`

**Review by allishultes — APPROVED** [review]
Looks good Eimi, and works without throwing a bug locally.

**Comment by emettely** [comment]
This needs another review before it can be merged

**Review by tamsingreen — COMMENTED** [review]
Works for me!

It looks like TranscriptTabContent and TranscriptEditor make use of a trackEvent prop but it aren't in the propTypes definition, would be good to add those too.

**Pull request merged**
